mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageMapper.java

```diff
@@ -140,33 +139,34 @@ public MailboxCounters getMailboxCounters(Mailbox mailbox) throws MailboxExcepti
     @Override
     public void delete(Mailbox mailbox, MailboxMessage message) {
         deleteAsFuture(message)
-            .join();
+            .block();
     }
 
-    private CompletableFuture<Void> deleteAsFuture(MailboxMessage message) {
+    private Mono<Void> deleteAsFuture(MailboxMessage message) {
         ComposedMessageIdWithMetaData composedMessageIdWithMetaData = message.getComposedMessageIdWithMetaData();
 
         return deleteUsingMailboxId(composedMessageIdWithMetaData);
     }
 
-    private CompletableFuture<Void> deleteUsingMailboxId(ComposedMessageIdWithMetaData composedMessageIdWithMetaData) {
+    private Mono<Void> deleteUsingMailboxId(ComposedMessageIdWithMetaData composedMessageIdWithMetaData) {
         ComposedMessageId composedMessageId = composedMessageIdWithMetaData.getComposedMessageId();
         CassandraMessageId messageId = (CassandraMessageId) composedMessageId.getMessageId();
         CassandraId mailboxId = (CassandraId) composedMessageId.getMailboxId();
         MessageUid uid = composedMessageId.getUid();
-        return CompletableFuture.allOf(
-            imapUidDAO.delete(messageId, mailboxId),
-            messageIdDAO.delete(mailboxId, uid)
-        ).thenCompose(voidValue -> indexTableHandler.updateIndexOnDelete(composedMessageIdWithMetaData, mailboxId));
+        return Flux.merge(
+                Mono.fromCompletionStage(imapUidDAO.delete(messageId, mailboxId)),
+                Mono.fromCompletionStage(messageIdDAO.delete(mailboxId, uid)))
+            .concatWith(Mono.fromCompletionStage(indexTableHandler.updateIndexOnDelete(composedMessageIdWithMetaData, mailboxId)))
+            .last();
     }
 
     @Override
     public Iterator<MailboxMessage> findInMailbox(Mailbox mailbox, MessageRange messageRange, FetchType ftype, int max) throws MailboxException {
         CassandraId mailboxId = (CassandraId) mailbox.getMailboxId();
         return retrieveMessages(retrieveMessageIds(mailboxId, messageRange), ftype, Limit.from(max))
-            .join()
             .map(SimpleMailboxMessage -> (MailboxMessage) SimpleMailboxMessage)
-            .sorted(Comparator.comparing(MailboxMessage::getUid))
+            .sort(Comparator.comparing(MailboxMessage::getUid))
```

Comment:
I really prefer materializing the Flux into collection when we want to sort because anyway it's what will happen with `sort`

Reply:
Do you have an example of that ?

Reply:
I'm not sure to understand the question, but I talk about using `collectSortedList`